src/PaddedViews.jl

```diff
@@ -219,7 +219,23 @@ julia> a1p[CartesianIndices(a1)]
 """
 function paddedviews(fillvalue, As::AbstractArray...)
     inds = outerinds(As...)
-    map(A->PaddedView(fillvalue, A, inds), As)
+    dims1 = 1
+    dims2 = (1,2)
+    dims3 = 1:2
```

Comment:
To be clear, `PaddedView` handles arrays of arbitrary dimensions, for example, `As` can be a list of arrays with dimension N=4. I don't think the current code handles `N>2` case.

Reply:
@johnnychen94 Can the person provide the tuple in like in any random fashion..like can I provide it in following ways:
- (1,2,4,5,3) - I need sort and unique for this
- (1,2,3,4,5)- if it's continous and properly sorted,then it won't be a issue
- (1,1,1) - also this case
- (0,1,2,...) - also this case

Reply:
I'll follow the cat convention:


julia> cat([1,2], [3 4]; dims=(2,1))
3×3 Array{Int64,2}:
 1  0  0
 2  0  0
 0  3  4

Reply:
Also we get dimension mismatch issue when arrays data is returned,which is mosaicviews data handling issue but that does arise for some reason I don't get yet...though we won't be talking about it here,right?

Reply:
The dimension mismatch issue you mentioned in #48 is not very clear to me, they should be unrelated. My best guess is that once we consolidate this feature in the PaddedViews side, the issue got fixed automatically.

FYI, there's a PR (https://github.com/JuliaArrays/MosaicViews.jl/pull/31) related to the `_padded_cat` you're referring to, so no, I'm not going to check it until we get the current PaddedViews PR done.